### ISSUE TITLE
Avoid rerenders if atom value doesn't change

### ISSUE THREAD

**Issue by SukkaW** [dai-shi/use-atom#28]
The issue is similar to the one in the jotai: https://github.com/pmndrs/jotai/issues/1158#issuecomment-1126839533

<img width="809" alt="image" src="https://user-images.githubusercontent.com/40715044/170677235-efa32234-cfc3-43c4-afd8-fa72f34e78d0.png">

Is it possible to apply a fix similar to https://github.com/pmndrs/jotai/pull/1159, to prevent atoms with the same primitive value from unnecessary committing?

**Comment by dai-shi** [comment]
No, it's not possible because we **always** run update function in render phase. And, thus, side effects are not allowed in the update function.
